File: src/main/java/com/kmarzecki/communicator/repository/UserRepository.java
package com.kmarzecki.communicator.repository;

import com.kmarzecki.communicator.model.auth.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Set;

/**
 * User repository
 */
@Repository
public interface UserRepository extends JpaRepository<UserEntity, Integer> {
    /**
     * Check if a user with username exists
     * @param username user username
     * @return whether a user exists
     */
    boolean existsByUsername(String username);

    /**
     * Find user bu username
     * @param username user username
     * @return User
     */
    UserEntity findByUsername(String username);

    /**
     * Find all users with usernames
     * @param usernames usernames
     * @return Set of users
     */
    Set<UserEntity> findAllByUsernameIn(Set<String> usernames);
}

File: src/main/kotlin/com/kmarzecki/trollo/repository/UserRepository.kt
package com.kmarzecki.trollo.repository

import com.kmarzecki.trollo.model.auth.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

/**
 * User repository
 */
@Repository
interface UserRepository : JpaRepository<UserEntity?, Int?> {
    /**
     * Check if a user with username exists
     * @param username user username
     * @return whether a user exists
     */
    fun existsByUsername(username: String?): Boolean

    /**
     * Find user bu username
     * @param username user username
     * @return User
     */
    fun findByUsername(username: String): UserEntity?

    /**
     * Find all users with usernames
     * @param usernames usernames
     * @return Set of users
     */
    fun findAllByUsernameIn(usernames: Set<String?>?): Set<UserEntity?>?
}